File: backend/src/main/java/com/finki/backend/web/response/InviteLinkResponse.java
package com.finki.backend.web.response;

import com.finki.backend.core.domain.enums.InviteStatus;
import lombok.Builder;

import java.time.Instant;

@Builder
public record InviteLinkResponse(
        Long id,
        Long roomId,
        String token,
        InviteStatus status,
        Instant expiresAt,
        Integer maxUses,
        Integer useCount,
        UserResponse createdBy,
        Instant createdAt
) {
}

File: frontend/src/app/core/services/response/invite-link.response.ts
import { InviteStatus } from '../../models/invite-status.enum';
import { UserResponse } from './user.response';

export interface InviteLinkResponse {
  id: number;
  roomId: number;
  token: string;
  status: InviteStatus;
  expiresAt: string;
  maxUses: number;
  useCount: number;
  createdBy: UserResponse;
  createdAt: string;
}
